src/me/hkgumbs/tictactoe/main/java/Main.java

```diff
@@ -0,0 +1,41 @@
+package me.hkgumbs.tictactoe.main.java;
+
+import me.hkgumbs.tictactoe.main.java.simulation.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class Main {
+    private static final String USAGE = "usage: java -jar tictactoe-java.jar" +
+            " [--size <number>]" +
+            " [--padding <number>]" +
+            " [--human|--minimax]";
+
+    private static final Configuration[] CONFIGURATIONS = new Configuration[]{
+            new SizeConfiguration(),
+            /* size must be applied first */
+            new PlayersConfiguration(System.in, System.out),
+            new RulesConfiguration(System.out),
+            new FormatterConfiguration()
+    };
+
+    public static void main(String[] args) {
+        Simulation simulation = new DefaultSimulation();
+        List<String> arguments = new ArrayList<>(Arrays.asList(args));
+
+        try {
+            for (Configuration configuration : CONFIGURATIONS)
+                configuration.apply(arguments, simulation);
```

Comment:
This design is nice, because it means you can add new configurations pretty easily.

It might be nice to separate this logic into another class, because looks like it'd be useful to test (things like catching exceptions, what to do if arguments are empty, how the configurations are applied, etc.). And `main()` is usually pretty tricky to get comprehensive tests around.